### ISSUE TITLE
[Feature]: Allow users to setup their profile during the sign up flow

### ISSUE THREAD

**Issue by LagunaElectric** [appsmithorg/appsmith#30605]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

When the user signs up for the first time they should be able to fill out their profile during the sign up flow and set their profile picture and display name.

It would also be acceptable to allow admins to set the users display name during the invite process.

### Why should this be worked on?

When a user signs up for the first time their display name is chosen for them. We deploy our apps with the navigation bar hidden, and because of this our users are unable to edit their profile to change their display name. We're making use of this display name in our UI.

**Comment by mohanarpit** [comment]
Unfortunately, I don't think this is a good idea. It introduces extra steps in the invite flow and/or the sign up flow. The more steps that are introduced, the lesser the likelihood of success. 

Since the user seems to be embedding the application (or deploying with a hidden top bar), I would modify the ask here to leverage Appsmith APIs using the session cookie to modify the user's name. This is also because if the user never sees the Appsmith top bar & then they suddenly see Appsmith's profile edit screen, it's a jarring experience that reduces trust for the user in the application they are using. 

TL;DR: We should not do this & instead, the developer should add leverage Appsmith APIs to edit the user's profile name with a custom screen.
